example/image-classification/common/fit.py

```diff
@@ -155,9 +159,16 @@ def fit(args, network, data_loader, **kwargs):
     head = '%(asctime)-15s Node[' + str(kv.rank) + '] %(message)s'
     logging.basicConfig(level=logging.DEBUG, format=head)
     logging.info('start with arguments %s', args)
+    
+    epoch_size = get_epoch_size(args, kv)
 
     # data iterators
     (train, val) = data_loader(args, kv)
+    if 'dist' in args.kv_store and 'sync' in args.kv_store:
```

Comment:
Is this also required for "async"?

Reply:
async should be good because no machine waits for other machines to pull gradients.

Reply:
so there's no need to resize for async right?

Reply:
Please check "async" instead

Reply:
Good catch Haibin, fixed it!